core/hoverfly.go

```diff
@@ -220,12 +220,15 @@ func (hf *Hoverfly) processRequest(req *http.Request) *http.Response {
 	}
 
 	if result.PostServeActionInputDetails != nil {
+
+		journalIDChannel := make(chan string, 1)
```

Comment:
it's possible that this channel never get closed, because `go postServeAction.Execute` is wrapped in an if statement.

Reply:
yup, correct. Made the required changes.